hal/src/boron/network/sara_ncp_client.h

```diff
@@ -73,6 +73,7 @@ class SaraNcpClient: public CellularNcpClient {
     volatile NcpState ncpState_ = NcpState::OFF;
     volatile NcpState prevNcpState_;
     volatile NcpConnectionState connState_ = NcpConnectionState::DISCONNECTED;
+    bool muxerChangedState_ = false;
```

Comment:
Let's not introduce flags to track specific conditions where we already have states and a kind of a state machine to handle things. This approach has the tendency to get out of control (see our main system event loop as an example of that).

Reply:
I had thought about adding a `NcpConnectionState::DISCONNECTING` state, but it was not obvious what else that might impact.  The idea of this flag is just to know when the NcpConnectionState has been changed outside of the place where it should only be changed.  If we can resolve that and make sure state is changed only in connectionState() directly, we shouldn't need any extra flags.  I would argue we don't need any additional state machines to track this either (the fact that NcpConnectionState is updated in the muxer callback).